src/test/java/org/jabref/logic/specialfields/SpecialFieldsUtilsTest.java

```diff
@@ -69,4 +72,18 @@ public void syncSpecialFieldsFromKeywordCausesNoChangeWhenKeywordsAreEmpty() {
         List<FieldChange> changes = SpecialFieldsUtils.syncSpecialFieldsFromKeywords(entry, ',');
         assertFalse(changes.size() > 0);
     }
+
+    @Test
+    public void updateFieldRemovesSpecialFieldKeywordWhenKeywordSyncIsUsed() {
+        BibEntry entry = new BibEntry();
+        SpecialField specialField = SpecialField.PRINTED;
+        Keyword specialFieldKeyword = specialField.getKeyWords().get(0);
+        // Add the special field
+        SpecialFieldsUtils.updateField(specialField, specialFieldKeyword.get(), entry, true, true, ',');
+        // Remove it
+        List<FieldChange> changes = SpecialFieldsUtils.updateField(specialField, specialFieldKeyword.get(), entry, true, true, ',');
+        assertTrue(changes.size() > 0);
```

Comment:
Please avoid checking such things with size, instead directly create a list expected and use assertEquals